cachi2/core/package_managers/yarn/project.py

```diff
@@ -62,12 +65,26 @@ def registry_server_for_scope(self, scope: str) -> str:
 
         See: https://v3.yarnpkg.com/configuration/yarnrc#npmScopes
         """
-        return NotImplemented
+        registry = self._data.get("npmScopes", {}).get(scope, {}).get("npmRegistryServer")
+
+        return registry or self.registry_server
 
     @classmethod
     def from_file(cls, file_path: RootedPath) -> "YarnRc":
         """Parse the content of a yarnrc file."""
-        return NotImplemented
+        try:
+            with file_path.path.open("r") as f:
+                yarnrc_data = yaml.safe_load(f)
+        except yaml.error.MarkedYAMLError as e:
```

Comment:
Should it be specifically MarkedYAMLError? There's also a generic yaml.YAMLError

Reply:
Changed it to YAMLError.